### PR TITLE
python311Packages.coffea: 2024.1.0 -> 2024.2.1

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Description of changes

Changelog: https://github.com/CoffeaTeam/coffea/releases/tag/v2024.1.2

Currently fails to build because of broken dependency `correctionlib`.

cc @veprbl 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 281235` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.fsspec-xrootd</li>
    <li>python312Packages.fsspec-xrootd.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.coffea</li>
    <li>python311Packages.coffea.dist</li>
    <li>python311Packages.fsspec-xrootd</li>
    <li>python311Packages.fsspec-xrootd.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
@fabaff I have added the changes from your PR.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 281235` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.fsspec-xrootd</li>
    <li>python312Packages.fsspec-xrootd.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.coffea</li>
    <li>python311Packages.coffea.dist</li>
    <li>python311Packages.fsspec-xrootd</li>
    <li>python311Packages.fsspec-xrootd.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/coffea/default.nix
Comment (SomeoneSerge) [inline pkgs/development/python-modules/coffea/default.nix:1]:
```
       > FAILED tests/test_2349_growablebuffer_in_numba.py::test_from_data
       > =========== 58 failed, 2659 passed, 35 skipped, 1 xfailed in 29.76s ============
       > /nix/store/jrcd51baalazjfg8s5qjlpkpl9nmqiqx-stdenv-linux/setup: line 1582: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/7balrfdm1qvjkmw96mlgayr4nmrxgyrh-python3.11-awkward-2.6.1.drv'.
```

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 281235` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.fsspec-xrootd</li>
    <li>python312Packages.fsspec-xrootd.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.coffea</li>
    <li>python311Packages.coffea.dist</li>
    <li>python311Packages.fsspec-xrootd</li>
    <li>python311Packages.fsspec-xrootd.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 281235` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python312Packages.fsspec-xrootd</li>
    <li>python312Packages.fsspec-xrootd.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.coffea</li>
    <li>python311Packages.coffea.dist</li>
    <li>python311Packages.fsspec-xrootd</li>
    <li>python311Packages.fsspec-xrootd.dist</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
I think that this is ready to be merged.
`coffea` builds fine, whereas it fails on `master`.
No regression seems to be introduce by this patch (clean `nixpkgs-review`).

File: pkgs/development/python-modules/coffea/default.nix
Comment (SomeoneSerge) [reply pkgs/development/python-modules/coffea/default.nix:1]: Ok so awkward is just broken on master/this is irrelevant: https://hydra.nixos.org/build/249807273

**Pull request merged**